gateway-util-common/src/main/java/org/apache/knox/gateway/util/ConcurrentSessionVerifier.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.knox.gateway.util;
+
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ConcurrentSessionVerifier {
```

Comment:
Instead of using static methods and static fields, let's have a normal non static class with instance variables and non static methods. Constants like `privilegedUserConcurrentSessionLimit`/`privilegedUserConcurrentSessionLimit` should be kept as statics but `concurrentSessionCounter` should be non static.

Then we can create one instance of this class and use that at each places where needed. Either by creating an instance somewhere in GatewayServer's `main()` and passing that around or by making a singleton. I prefer the former but that's up to how the code is structured. We can decide that later when we start using it.